core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java

```diff
@@ -76,8 +79,12 @@ public String toString() {
     }
   }
 
-  // TODO(binfan): Add refcount to the UFS instance. Once the refcount goes to zero,
-  // we could close this UFS instance.
+  /**
+   * This map keeps track of the reference counter for an {@link UnderFileSystem}.
+   */
+  private final ConcurrentHashMap<Class<? extends UnderFileSystem>, AtomicLong> mReferenceCounter =
```

Comment:
One possible easy way to do so is to have `mUnderFileSystemMap` mapping from `Key` to a tupe of `<UnderFileSystem, AtomicLong>`